src/metabase/models/card.clj

```diff
@@ -305,6 +305,13 @@
   :in mi/json-in
   :out result-metadata-out)
 
+(defn- nil-dataset-query-to-dummy-query
+  [{dataset-query :dataset_query :as card}]
+  (cond-> card
+    (nil? dataset-query) (assoc :dataset_query {:database (:database_id card)
+                                                :type     :native
+                                                :native   {:query "select 'this query broke, sorry' x"}})))
```

Comment:
I don't love this approach really. I'd rather have things actually fail and not break the app, but this solution at least gets a corrupt card to 'load' (for some sense of the word).

If there is a way to reconstruct the actual query, that would be much cooler.

Obviously, I think there needs to be a way to more robustly prevent invalid json from entering the db in the first place, so that we don't need a weird post-select solution like this.